src/Avalonia.Controls/Shapes/Arc.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using Avalonia.Media;
+
+namespace Avalonia.Controls.Shapes
+{
+    public class Arc : Shape
+    {
+        /// <summary>
+        /// Defines the <see cref="StartAngle"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> StartAngleProperty =
+            AvaloniaProperty.Register<Arc, double>(nameof(StartAngle), 0.0);
+
+        /// <summary>
+        /// Defines the <see cref="SweepAngle"/> property.
+        /// </summary>
+        public static readonly StyledProperty<double> SweepAngleProperty =
+            AvaloniaProperty.Register<Arc, double>(nameof(SweepAngle), 0.0);
+
+        static Arc()
+        {
+            StrokeThicknessProperty.OverrideDefaultValue<Arc>(1);
+            AffectsGeometry<Arc>(BoundsProperty, StrokeThicknessProperty, StartAngleProperty, SweepAngleProperty);
+        }
+
+        /// <summary>
+        /// Gets or sets the angle at which the arc starts, in degrees.
+        /// </summary>
+        public double StartAngle
+        {
+            get { return GetValue(StartAngleProperty); }
+            set { SetValue(StartAngleProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the angle at which the arc ends relative to the start angle, in degrees.
+        /// </summary>
+        public double SweepAngle
+        {
+            get { return GetValue(SweepAngleProperty); }
+            set { SetValue(SweepAngleProperty, value); }
+        }
+        
+        protected override Geometry CreateDefiningGeometry()
+        {
+            double angle1 = DegreesToRad(StartAngle);
+            double angle2 = angle1 + DegreesToRad(SweepAngle);
+            
+            double startAngle = Math.Min(angle1, angle2);
+            double sweepAngle = Math.Max(angle1, angle2);
+
+            double normStart = RadToNormRad(startAngle);
+            double normEnd = RadToNormRad(sweepAngle);
+
+            var rect = new Rect(Bounds.Size);
+
+            if ((normStart == normEnd) && (startAngle != sweepAngle)) //complete ring
+            {
+                return new EllipseGeometry(rect.Deflate(StrokeThickness / 2));
+            }
+            else if (SweepAngle == 0)
+            {
+                return new StreamGeometry();
+            }
+            else //partial arc
```

Comment:
Format this comment

Reply:
...er...what exactly do you mean by that?

Reply:
I suppose it should be with a leading whitespace

```cs
else // partial arc
```